### DISCUSSION TITLE
What is the best way to manage app versioning across both app stores?

### DISCUSSION THREAD

**Discussion by engineerliz** [codemagic-ci-cd/codemagic-docs#1976] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. I've been using pattern `version: x.x.x+y` in pubspec.yaml to version the app. For each incremental update `y`, so for example `1.0.1+1` -> `1.0.1+2` 
2. On ios app store, this versioning patterns works as expected. However on google playstore, the versioning shows up as `y (x.x.x)`. below screenshot is for `version: 1.0.2+39`
<img width="250" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/111380693/0657f8dc-78d8-4ce1-b197-98d2c2987505">

3. I recently upgraded to `1.0.3+1` and now the playstore shows the version as `1 (1.0.3)` which it thinks is a lower version than `39 (1.0.2)` from above
<img width="384" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/111380693/bc1aaf00-35b4-4553-bb17-19945cf2bc45">
4. This is blocking codemagic from uploading the newest app version to playstore because playstore blocks "lower version app builds"

### Expected results

I would expect either the versioning in google playstore to correctly reflect the versioning pattern or for there to be an option in the code magic console to configure versioning. 

### Actual results

Right now I do not see a way to fix the current versioning situation for 1.0.3+1

**Comment by himesh-cm** [comment]
Hi @engineerliz, I suggest checking the docs [here](https://docs.codemagic.io/knowledge-codemagic/build-versioning/) with the docs on Auto build versioning.

**Reply by engineerliz** [reply]
thanks fro the quick response! the doc you linked shows how to do it via .yaml but is there a way to configure this using the workflow editor?

**Reply by himesh-cm** [reply]
You need to append the command to the Flutter build command, it can be `yaml` or `workflow editor`
`--build-name=1.0.$PROJECT_BUILD_NUMBER --build-number=$PROJECT_BUILD_NUMBER`.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
